n3fit/src/n3fit/layers/DY.py

```diff
@@ -94,10 +95,12 @@ def compute_dy_observable_one_replica(pdf, mask_and_fk):
     masking the PDF rather than the fk table.
     """
     mask, fk = mask_and_fk
-    pdf = pdf[0][0]  # yg
+    pdfb = pdf[0][0][0]  # yg
+    pdfa = pdf[1][0][0]  # yg
```

Comment:
```suggestion
    # Retrieve the two PDFs (which may potentially be coming from different initial states)
    # Since this is the one-replica function, remove the batch and replica dimension
    pdfb = pdf[0][0][0]  # xf
    pdfa = pdf[1][0][0]  # yg
```